builtin/mainmenu/textures.lua

```diff
@@ -145,6 +145,27 @@ function mm_texture.set_game(identifier,gamedetails)
 	
 	local path = gamedetails.path .. DIR_DELIM .."menu" ..
 									 DIR_DELIM .. identifier .. ".png"
+
+	if core.setting_getbool("random_menu_images") == true then
+		local count = 1
+		for i = 1, 9 do
+			local check_path = path .. "." .. i
+			local file = io.open(check_path,"r")
+			if file == nil then
+				count = i - 1
+				break
+			else
+				io.close(file)
+			end
+		end
+		local new_path = path .. "." .. math.random(1, count)
+		local exists = io.open(new_path,"r")
+		if exists ~= nil and math.random(1, (count+1)) ~= 1 then
```

Comment:
That is what I intended it to do. The first math.random is to determine which of the random screenshots will be used, the second (the second value of which is one higher than the second value of the first math.random) is to determine whether the first one is used or not. So it doesn't matter whether the values produced are different.